torchao/sparsity/README.md

```diff
@@ -20,29 +20,32 @@ More concretely, we hope to provide tutorials and APIs for both sparse kernels (
 
-The inference acceleration of semi-structured sparsity depends on the matmul shapes, which is why we don't see additional speedups when applying to all linear layers (attn + mlp) of segment-anything.
-We find that accelerating the MLP linear layers provied the most speedups (`lin1`, `lin2`). To repoduce our benchmarks you can run the following command:
+Overall, we found that accelerating the MLP linear layers provied the most speedups (`lin1`, `lin2`), while mitigating accuracy loss.
 
-```
-python benchmarks/benchmark_sam.py
-```
+Applying sparsity to the attention linear layers led to a slower model, likely due to two reasons:
+- We cannot fuse into our semi-structured sparse matmul with torch.compile.
+- The speedups we observe for sparse matmul depend on the matmul shapes, and the attention matmuls are smaller than the MLP ones.
+
+We were also are able to compose int8 dynamic quantization with 2:4 sparsity for futher speedups.
+
+We found that applying int8 dynamic quantization to the attention layers, int8 dynamic quantization + 2:4 sparsity to mlp layer 1 and 2:4 sparsity to mlp layer 2 yielded the best configuration.
+
+The following benchmarks we ran for sam ViT-h on an NVIDIA-A100-80GB, with batch_size=32 and `bfloat16` dtype, with `torch.compile="max_autotune"`:
 
-The following benchmarks we run on an A100, with batch_size=32 and `bfloat16` dtype:
+| Model Type | Technique                                                                                            | img/s | memory (MiB) | mIoU (coco2017 val) | relative speedup | relative accuracy |
+|------------|------------------------------------------------------------------------------------------------------|-------|--------------|---------------------|------------------|-------------------|
+| ViT-h      | baseline (bfloat16, max-autotune)                                                                    | 22.75 | 15172        | 0.5811              |                  |                   |
+|            | int8 dynamic quant (attn + mlp)                                                                      | 24.91 | 15154        | 0.5822              | **1.09x**        | **100.19%**       |
+|            | 2:4 sparsity (mlp only)                                                                              | 24.81 | 15632        | 0.5672              | **1.10x**        | **97.61%**        |
+|            | 2:4 sparsity (attn + mlp)                                                                            | 24.30 | 13429        | 0.5306              | **1.07x**        | **91.31%**        |
+|            | int8 dynamic quant (attn)<br>int8 dynamic quant + 2:4 sparsity (mlp lin1)<br>2:4 sparsity (mlp lin2) | 26.46 | 14865        | 0.5668              | **1.16x**        | **97.54%**        |
 
-| qkv  | proj | lin1 | lin2 | time | memory | img/s |
-| ---- | ---- | ---- | ---- | ---- | ------ | ----- |
-| None | None | None | None | 1361.73 | 15.81 | 23.50 |
```

Comment:
These numbers are a bit higher than the one reported in the new benchmark script, I'm pretty sure this is because we reuse the same example for benchmarking and not because of any changes to the code. 

I grabbed the output of TORCH_LOGS=output_code for both the new and old benchmark script: 

<img width="1152" alt="Screenshot 2024-06-27 at 10 43 56 PM" src="https://github.com/pytorch/ao/assets/8041643/12af5043-b71f-4dd3-a640-dc1ec15d3be4">